### ISSUE TITLE
Creating multiple entry in auto.master file for single automaster_entry '/-'

### ISSUE THREAD

**Issue by manish-nuance** [sous-chefs/autofs#40]
### :speaking_head: Foreword

Multiple entry created in `auto.master` file for single `automaster_entry` resource.
Thank for taking the time to fill this bug report fully. Without it we may not be able to fix the bug, and the issue may be closed without resolution.

## :ghost: Brief Description

My recipe configurations:
```ruby
automaster_entry '/-' do
  map '/etc/auto.top'
  options '--timeout 120'
end
```
It does create two entry in auto.master file   -- observed bugs
```
/- /etc/auto.top --timeout 120
/top /etc/auto.top
```
It should be single line entry on auto.master file
```
/- /etc/auto.top --timeout 120
```
## :pancakes: Cookbook version

Version of the cookbook where you are encountering the issue.

## :woman_cook:  Chef-Infra Version
tag versio 2.9.1
Version of chef-client in your environment.
chef 15.9.17
## :tophat: Platform details
 Operating system distribution and release version. Cloud provider if running in the cloud.
CentOS-7.7

## Steps To Reproduce
Configure chef recipe with following..
My recipe configurations:
```ruby
automaster_entry '/-' do
  map '/etc/auto.top'
  options '--timeout 120'
end
```
## :police_car: Expected behavior

should come-up with single entry in auto.master
```
/- /etc/auto.top --timeout 120 
```
## :heavy_plus_sign: Additional context

Add any other context about the problem here. e.g. related issues or existing pull requests.

**Comment by ramereth** [comment]
@manish-nuance I'm not able to reproduce this issue. Can you please let me know what version of the line cookbook you're using?

**Comment by manish-nuance** [comment]
@ramereth , thanks for taking this issue forward..
I'm using following tag version of autofs and line
autofs-2.3.0
line-2.9.1

thanks..

**Comment by ramereth** [comment]
Unfortunately I'm still not able to replicate this issue. Are you able to replicate this with kitchen on your end? I'm wondering if the issue is cased by a specific line in your `/etc/auto.master` that isn't in the default config. Would it be possible to provide your file before running this resource so I can verify?

**Comment by manish-nuance** [comment]
facing multiple entry issue with direct maps like "/-".. I'm using custom chef recipe to apply following.
did your try with followings.. ?

automaster_entry '/-' do
  map '/etc/auto.top'
  options '--timeout 120'
end

this automaster_entry "/-" gives multiple entry in auto.master
/- /etc/auto.top --timeout 120   ;; this line of entry is good
/top /etc/auto.top   ;; this line of entry is created with map files extension.

Otherwise,  look all good with indirect maps like "/xyz"

**Comment by ramereth** [comment]
Yes I tried that and couldn't produce this creating a duplicate entry. That's why I'm wondering if there is something else in the file that the regex is having an issue with that I need on my end so I can replicate it.
